torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,375 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Dataset splitting utilities."""
+
+from copy import deepcopy
+from math import floor
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+from rtree.index import Index, Property
+from torch import Generator, default_generator, randint, randperm
+from torch._utils import _accumulate
```

Comment:
Is there another option we could use here that is public?

Reply:
changed for itertools' accumulate